xlators/storage/posix/src/posix-entry-ops.c

```diff
@@ -1184,6 +1196,42 @@ posix_unlink_gfid_handle_and_entry(call_frame_t *frame, xlator_t *this,
     return -1;
 }
 
+static int
+posix_unlink_stale_linkto(call_frame_t *frame, xlator_t *this,
+                          const char *real_path, int32_t *op_errno, loc_t *loc)
+{
+    int ret = -1;
+    struct iatt stbuf = {
+        0,
+    };
+
+    /* get stale file gfid and stat-info */
+    ret = posix_pstat(this, NULL, NULL, real_path, &stbuf, _gf_false);
+    if (ret) {
+        if (errno == ENOENT)
+            ret = 0; /* retry creation if file was deleted */
+        gf_msg(this->name, GF_LOG_INFO, 0, P_MSG_LSTAT_FAILED,
+               "lstat on %s failed", real_path);
+        goto out;
+    }
+    /* unlink only if linkto filw*/
+    if (IS_DHT_LINKFILE_MODE(&stbuf)) {
+        gf_msg(this->name, GF_LOG_INFO, 0, P_MSG_HANDLE_CREATE,
+               "unlinking stale linkto: %s gfid: %s", real_path,
+               uuid_utoa(stbuf.ia_gfid));
+        ret = posix_unlink_gfid_handle_and_entry(
+            frame, this, real_path, &stbuf, op_errno, loc, _gf_false, NULL);
+
+    } else {
+        gf_msg(this->name, GF_LOG_WARNING, 0, P_MSG_HANDLE_CREATE,
```

Comment:
Ok for debugging, but I think it should be removed in the final patch (or keep it in debug level).